pkg/chunk/cache/background.go

```diff
@@ -85,9 +87,9 @@ func (c *backgroundCache) Store(ctx context.Context, key string, buf []byte) err
 	}
 	select {
 	case c.bgWrites <- bgWrite:
-		queueLength.Inc()
+		queueLength.WithLabelValues(c.name).Inc()
```

Comment:
Hmm, can we hold the `Guage`, `Counter` in `backgroundCache` itself, instead of calling `WithLabelValues` everytime?

i.e,
```
func NewBackground(name string, cfg BackgroundConfig, cache Cache) Cache {
	c := &backgroundCache{
		Cache:    cache,
		quit:     make(chan struct{}),
		bgWrites: make(chan backgroundWrite, cfg.WriteBackBuffer),
		name:     name,
                 queueLength: queueLength.WithLabelValues(name),
                 droppedWriteBack: droppedWriteBack.WithLabelValues(name),
	}
...
...
```

Reply:
Yea, I can update it